### ISSUE TITLE
Connection error on exists

### ISSUE THREAD

**Issue by CrabbyPete** [dask/s3fs#384]
The files in the directory are huge. Some are a > 1G. 
**What happened**:
Connection timed out
**What you expected to happen**:

**Minimal Complete Verifiable Example**:

```python
# Put your MCVE code here
```

**Anything else we need to know?**:

**Environment**:

- Dask version:
- Python version:3.4
- Operating System:linux
- Install method (conda, pip, source):pip

**Comment by martindurant** [comment]
Sorry, I don't have any suggestions from that information.
I do notice that your python is rather old, so you might want to start with upgrading to at least py3.7 .

**Comment by CrabbyPete** [comment]
Unfortunately the server I am working on only has 3.4. Don't you support 3.4? This also happens with a S3.put if the file is larger than 1G

**Comment by martindurant** [comment]
No, python 3.4 is out of its support period, as is 3.5 ( 3.6 was released 2016-12-23; https://en.wikipedia.org/wiki/History_of_Python#Table_of_versions ), and the async implementation requires syntax that did not exist back then. You can, however, try older versions of s3fs.
